tests/apollo-fetch.ts

```diff
@@ -0,0 +1,496 @@
+
+const { assert, expect } = chai;
+
+const sampleQuery = gql`
+query SampleQuery {
+  stub{
+    id
+  }
+}
+`;
+
+describe('apollo-fetch', () => {
+  const postData = {hello: 'world', method: 'POST'};
+  const data = JSON.stringify({data: { hello: 'world' }});
+  const unparsableData = 'raw string';
+  const mockError = { throws: new TypeError('mock me') };
+  const swapiUrl = 'http://graphql-swapi.test/';
+  const missingUrl = 'http://does-not-exist.test/';
+
+  const unauthorizedUrl = 'http://unauthorized.test/';
+  const serviceUnavailableUrl = 'http://service-unavailable.test/';
+
+  const simpleQueryWithNoVars = gql`
+    query people {
+      allPeople(first: 1) {
+        people {
+          name
+        }
+      }
+    }
+  `;
+
+  const simpleQueryWithVar = gql`
+    query people($personNum: Int!) {
+      allPeople(first: $personNum) {
+        people {
+          name
+        }
+      }
+    }
+  `;
+
+  const simpleResult = {
+    data: {
+      allPeople: {
+        people: [
+          {
+            name: 'Luke Skywalker',
+          },
+        ],
+      },
+    },
+  };
+
+  const complexQueryWithTwoVars = gql`
+    query people($personNum: Int!, $filmNum: Int!) {
+      allPeople(first: $personNum) {
+        people {
+          name
+          filmConnection(first: $filmNum) {
+            edges {
+              node {
+                id
+              }
+            }
+          }
+        }
+      }
+    }
+  `;
+
+  const complexResult = {
+    data: {
+      allPeople: {
+        people: [
+          {
+            name: 'Luke Skywalker',
+            filmConnection: {
+              edges: [
+                {
+                  node: {
+                    id: 'ZmlsbXM6MQ==',
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    },
+  };
+
+  before(() => {
+
+    fetchMock.post('/graphql', data);
+    fetchMock.post('/raw', unparsableData);
+    fetchMock.post('begin:data', postData);
+    fetchMock.post('begin:error', mockError);
+    fetchMock.post('test', data);
+
+    fetchMock.post(swapiUrl, (url, opts) => {
+      const { query, variables } = JSON.parse((opts as RequestInit).body!.toString());
+
+      if (query === print(simpleQueryWithNoVars)) {
+        return simpleResult;
+      }
+
+      if (query === print(simpleQueryWithVar)
+          && isEqual(variables, { personNum: 1 })) {
+        return simpleResult;
+      }
+
+      if (query === print(complexQueryWithTwoVars)
+          && isEqual(variables, { personNum: 1, filmNum: 1 })) {
+        return complexResult;
+      }
+      throw new Error('Invalid Query');
+    });
+    fetchMock.post(missingUrl, () => {
+      throw new Error('Network error');
+    });
+
+    fetchMock.post(unauthorizedUrl, 403);
+    fetchMock.post(serviceUnavailableUrl, 503);
+  });
+
+  afterEach(fetchMock.reset);
+
+  it('should not throw with no arguments', () => {
+    assert.doesNotThrow(createApolloFetch);
```

Comment:
For people who don't want to provide any options, do we have a simpler function they could import? e.g. `apolloFetch`?